packages/amqp/lib/AbstractAmqpPublisher.ts

```diff
@@ -42,6 +44,22 @@ export abstract class AbstractAmqpPublisher<MessagePayloadType extends object>
     }
     resolveSchemaResult.result.parse(message)
 
+    // If it's not initted yet, do the lazy init
+    if (!this.isInitted) {
+      // avoid multiple concurrent inits
+      if (!this.initPromise) {
+        this.initPromise = this.init()
+      }
+      this.initPromise
```

Comment:
it is intentional that this promise is not awaited, that's how we keep the method invocation synchronous

RabbitMQ publish by itself doesn't guarantee that your message is delivered successfully, so this kind of fire-and-forget is not strongly different from how amqp-lib behaves in the first place.

Reply:
moved to comment